js/commandRegistry.js

```diff
@@ -0,0 +1,59 @@
+/**  TableMapper Command Registry
+  * ---------------------------
+  * A registry of all available commands in the
+  * TableMapper ecosystems.
+  */
+
+const commandRegistry = {
+    "copy" : _onCopyCommand,
+}
+
+/* the commands */
+function _onCopyCommand(source, target, arg){
+    console.log(`going to copy ${source} to ${target}`);
+    const sourceSheet = document.getElementById("source");
+    const targetSheet = document.getElementById("target");
+    const [sourceOrigin, sourceCorner] = source.split(":");
+    const [sourceOriginX, sourceOriginY] = parseCoordinates(sourceOrigin);
+    const [sourceCornerX, sourceCornerY] = parseCoordinates(sourceCorner);
+    const [targetOriginX, targetOriginY] = parseCoordinates(target);
+    // iterate over the coordinates and fill in target sheet data values
+    let currentX = sourceOriginX;
+    let currentY = sourceOriginY;
+    let targetX = targetOriginX;
+    let targetY = targetOriginY;
+    while (currentX <= sourceCornerX){
+        while(currentY <= sourceCornerY){
+            let value = sourceSheet.dataFrame.getAt([currentX, currentY]);
+            // the presence of an arg tell us that we should apply it to the string
+            // in the JS way
+            if(arg){
+                value = eval(`'${value}'.${arg}`);
+            }
+            targetSheet.dataFrame.putAt([targetX, targetY], value);
```

Comment:
We will want to explicitly pass `false` as the third argument here. That prevents `putAt()` from telling the sheet ti dispatch a `data-changed` event each time. Instead, we can tell it to dispatch that event only once, at the end of the function. It can be accomplished with something like the following:
```javascript
if(this.dataFrame.callback){
    this.dataFrame.callback(// some frame here)
}
```
where the Frame we pass is a Frame instance representing the data-relative frame that was updated in the target sheet.